tensorrt_llm/_torch/modules/fused_moe/mega_moe/mega_moe_deepgemm.py

```
@@ -55,7 +55,122 @@
+    cached, cached_pg = entry
+    if cached_pg is ep_pg:
+        return cached
+    del _MEGA_MOE_SYMM_BUFFER_CACHE[key]
+    # This runs on the next LLM's init path and the stale buffer was
+    # rendezvoused over an already-destroyed group, so a raising free must
+    # not escape and fail the new LLM -- fall through to a fresh allocation
+    # instead (no worse than the pre-fix behavior, which leaked and
+    # allocated).
+    try:
+        freed = _free_symm_buffer(cached)
+    except Exception as e:
```

Ruff (0.16.2)

[warning] Do not catch blind exception: `Exception`

(BLE001)